Documentation/Relationship/Relationship.md

```diff
@@ -0,0 +1,598 @@
+} content: {
+  Handler()
+}
+```
+
+### 2.1. Overriding the relationship name
+
+The in chapter [1.](#1-using-the-dsl-structure) described inference approach uses the string 
+description of the `PathComponent` as the relationship name. This may not fit everybody's needs.
+
+Such a customization mechanism is provided with the `.relationship` modifier (with the external parameter `name`):
+
+```swift
+var content: some Component {
+  Group {
+    "test".relationship(name: "new-name")
+  } content: {
+    Handler()
+  }
+}
+```
+
+The same `PathComponent` name cannot be used multiple times on the same depth in the `PathComponent` tree.
+Similarly the renamed Relationship MUST NOT collide with any other relationship name on the same depth.
+
+### 2.2. Adding new relationships
+
+To support rearranging components without breaking HATEOAS linking information or the GraphQL query schema,
+a user can also **add** their own relationship definitions.
+
+As we don't want users to manually list every `PathComponent` for the destination of the relationship, the user can create
+instances of `Relationship` which are then referenced with `.relatable` (marking destination) and
+`.relationship(to:)` modifiers.
+
+Those `Relationship` instances can be defined anywhere, preferably as a property of a `Component`.  
+Elsewise `Relationship` instances can also be organized into `RelationshipContainer` which are collected through
+the `relationshipContainer` property of a `Component`. Above modifiers then also accept KeyPaths into such containers.
+
+The example below showcases both ways of using `Relationship` instances:
+```swift
+struct TestServiceRelationShips: RelationshipContainer {
+    var greeter = Relationship("greeter")
+}
+
+struct TestService: WebService {
+
+  @PathParameter var userId: User.ID
+
+  var relationshipContainer = TestServiceRelationShips()
+  let otherRelationship = Relationship("other") 
+
+  var content: some Component {
+    Group {
+      "user"
+      $userId
+        .relationship(to: \TestServiceRelationShips.greeter)
+        .relationship(to: otherRelationship)
+    } content: {
+      Handler()
+    }
+    Group("greeting", $userId) {
+      Greeter()
+        .relatable(\TestServiceRelationShips.greeter)
+    }
+    Group("other") {
+      Other()
+        .relatable(otherRelationship)
+    }
+  }
+}
+```
+
+### 2.3. Hiding Relationships
+
+Specifically for the REST Interface Exporter it may be a sensible decision for a user to hide relationship information
+generated by Apodini.
+
+This is made possible by introducing a `.hideLink` modifier.
+
+The GraphQL Interface Exporter MUST ignore the `hidden` flag and treat it like any other relationship.
+
+Considering the example below, a request to `/` will not include a relationship pointing to `/test`,
+but the endpoint `/test` will nonetheless be reachable when requesting it directly.
+
+```swift
+var content: some Component {
+  Group {
+    "test".hideLink()
```

Comment:
I would suggest to use a common naming across all modifiers. E.g. either relationship, reference or link.

Reply:
Not using relationship here was intended. I thought using `hideRelationship` is a bit misleading, as GraphQL will ignore the `hidden` flag and the REST exporter is really the only one using it. So the idea was to make that clear by using different naming here. But maybe there is a better word than `hide`?

Reply:
I see. Reading through your explanation I see why `link` would be desirable in this context and I would even prefer `hideLink` over `hideRelationship`. We should somehow document the reasoning behind either choice we make as part of the proposal.

Reply:
Resolved. See https://github.com/Apodini/Apodini/pull/41#issuecomment-748679123.